### DISCUSSION TITLE
Missing interface in foxglove snap for Gamepad.

### DISCUSSION THREAD

**Discussion by Guillaumebeuzeboc** [foxglove/community#980] [Bugs] [answered]
Previously discussed here: https://github.com/foxglove/studio/issues/7409
With foxglove-studio snap, the [joystick extension](https://github.com/joshnewans/foxglove-joystick) required the joystick interface to access the joystick.
The extension got added, but after a test it's still not working.
When trying to attach a gamepad, we get the following log:
```
Failed to begin udev enumeration.
```

It appears that Chrome [getGamepads](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/getGamepads) function requires the access to `/run/udev/data/` that would be provided by the [`hardware-observe`](https://snapcraft.io/docs/hardware-observe-interface) interface.
Once added to the snap, one can connect it with:
```
sudo snap connect foxglove-studio:hardware-observe
```
Hopefully that should be enough for the extension to work properly in the snap.

Foxglove snap version: 2.0.1

**Comment by achim-k** [comment]
I guess we can add that. Does the official chromium snap also have this interface?

**Comment by Guillaumebeuzeboc** [comment]
Yes it does, it can be checked with:
```
sudo snap connections chromium | grep hardware-observe
```

**Reply by Guillaumebeuzeboc** [reply]
It even got the autoconnect request here: https://forum.snapcraft.io/t/chromium-hardware-observe-auto-connect-request/37742

**Comment by achim-k** [comment, marked as answer]
We have added the `hardware-observe` interface. It will be available with the next snap release.
